### PR TITLE
Hw 03 dao 09

### PR DESCRIPTION
**Pull request by Alex-Korzun**

### CODE REVIEW

File: src/main/webapp/web.xml [hunk 1]
```diff
@@ -24,70 +24,70 @@
 
     <servlet>
         <servlet-name>createDriver</servlet-name>
-        <servlet-class>manufacturing.controllers.drivers.CreateDriverController</servlet-class>
+        <servlet-class>manufacturing.controller.driver.CreateDriverController</servlet-class>
     </servlet>
     <servlet-mapping>
         <servlet-name>createDriver</servlet-name>
-        <url-pattern>/drivers/create</url-pattern>
+        <url-pattern>/driver/create</url-pattern>
```
Comment (gigbat) [inline src/main/webapp/web.xml]:
```suggestion
        <url-pattern>/drivers/create</url-pattern>
```

File: src/main/webapp/web.xml [hunk 2]
```diff
@@ -24,70 +24,70 @@
...
 
     <servlet>
         <servlet-name>createManufacturer</servlet-name>
-        <servlet-class>manufacturing.controllers.manufacturers.CreateManufacturerController</servlet-class>
+        <servlet-class>manufacturing.controller.manufacturer.CreateManufacturerController</servlet-class>
     </servlet>
     <servlet-mapping>
         <servlet-name>createManufacturer</servlet-name>
-        <url-pattern>/manufacturers/create</url-pattern>
+        <url-pattern>/manufacturer/create</url-pattern>
```
Comment (gigbat) [inline src/main/webapp/web.xml]: same and check every url-pattern

**Review by gigbat — APPROVED** [review]

File: src/main/webapp/web.xml (hunk 1 above)
Comment (Alex-Korzun) [reply src/main/webapp/web.xml]: done

File: src/main/webapp/web.xml (hunk 2 above)
Comment (Alex-Korzun) [reply src/main/webapp/web.xml]: done everywhere

File: src/main/java/manufacturing/security/impl/AuthenticationServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+    private DriverService driverService;
+
+    @Override
+    public Driver login(String login, String password) throws AuthenticationException {
+        Driver driverFromDB = driverService.findByLogin(login).orElseThrow(() ->
+                new AuthenticationException("Incorrect login or password"));
+        if (driverFromDB.getPassword().equals(password)) {
+            return driverFromDB;
+        }
+        throw new AuthenticationException("Incorrect login or password");
```
Comment (Sofasmile) [inline src/main/java/manufacturing/security/impl/AuthenticationServiceImpl.java]:
```suggestion
        Optional<Driver> driver = driverService.findByLogin(login);
        if (driver.isPresent() && driver.get().getPassword().equals(password)) {
            return driver.get();
        }
        throw new AuthenticationException("Incorrect login or password");
```

File: src/main/java/manufacturing/web/filter/AuthenticationFilter.java [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+
+    }
+
+    @Override
+    public void doFilter(ServletRequest servletRequest, ServletResponse servletResponse,
+                         FilterChain filterChain) throws IOException, ServletException {
+        HttpServletRequest req = (HttpServletRequest) servletRequest;
+        HttpServletResponse resp = (HttpServletResponse) servletResponse;
+        String url = req.getServletPath();
+        if (url.equals("/login") || url.equals("/drivers/create")) {
```
Comment (Sofasmile) [inline src/main/java/manufacturing/web/filter/AuthenticationFilter.java]: read slack

File: src/main/java/manufacturing/web/filter/AuthenticationFilter.java [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+            filterChain.doFilter(req, resp);
+            return;
+        }
+        Long driverId = (Long) req.getSession().getAttribute(DRIVER_ID);
+        if (driverId == null || driverService.get(driverId) == null) {
```
Comment (Sofasmile) [inline src/main/java/manufacturing/web/filter/AuthenticationFilter.java]: https://mate-academy.github.io/jv-program-fulltime/02_jdbc/checklist/09_filters_checklist.html
- 5 paragraph

**Review by Sofasmile — CHANGES_REQUESTED** [review]

File: src/main/java/manufacturing/security/impl/AuthenticationServiceImpl.java (hunk 3 above)
Comment (Alex-Korzun) [reply src/main/java/manufacturing/security/impl/AuthenticationServiceImpl.java]: done

File: src/main/java/manufacturing/web/filter/AuthenticationFilter.java (hunk 4 above)
Comment (Alex-Korzun) [reply src/main/java/manufacturing/web/filter/AuthenticationFilter.java]: read. done

File: src/main/java/manufacturing/web/filter/AuthenticationFilter.java (hunk 5 above)
Comment (Alex-Korzun) [reply src/main/java/manufacturing/web/filter/AuthenticationFilter.java]: sorry. didn't understand this check at first. done

File: src/main/java/manufacturing/web/filter/AuthenticationFilter.java [hunk 6]
```diff
@@ -17,10 +19,13 @@
     private static final Injector injector = Injector.getInstance("manufacturing");
     private final DriverService driverService =
```
Comment (Sofasmile) [inline src/main/java/manufacturing/web/filter/AuthenticationFilter.java]: do we need these fields?

**Review by Sofasmile — CHANGES_REQUESTED** [review]

File: src/main/java/manufacturing/web/filter/AuthenticationFilter.java (hunk 6 above)
Comment (Alex-Korzun) [reply src/main/java/manufacturing/web/filter/AuthenticationFilter.java]: no. removed

**Review by Sofasmile — APPROVED** [review]

**Pull request merged**